subsys/debug/thread_awareness.c

```diff
@@ -0,0 +1,119 @@
+	/* Version 0 starts */
+	[ZTA_OFFSET_VERSION] = 1,
+	[ZTA_OFFSET_K_CURR_THREAD] = offsetof(struct _cpu, current),
+	[ZTA_OFFSET_K_THREADS] = offsetof(struct z_kernel, threads),
+	[ZTA_OFFSET_T_ENTRY] = offsetof(struct k_thread, entry),
+	[ZTA_OFFSET_T_NEXT_THREAD] = offsetof(struct k_thread, next_thread),
+	[ZTA_OFFSET_T_STATE] = offsetof(struct _thread_base, thread_state),
+	[ZTA_OFFSET_T_USER_OPTIONS] = offsetof(struct _thread_base,
+						   user_options),
+	[ZTA_OFFSET_T_PRIO] = offsetof(struct _thread_base, prio),
+#if defined(CONFIG_ARM64)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.sp),
+#elif defined(CONFIG_ARM)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.psp),
+#elif defined(CONFIG_ARC)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.sp),
+#elif defined(CONFIG_X86)
+#if defined(CONFIG_X86_64)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.rsp),
+#else
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.esp),
+#endif
+#elif defined(CONFIG_NIOS2)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.sp),
+#elif defined(CONFIG_RISCV)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.sp),
+#elif defined(CONFIG_SPARC)
+	[ZTA_OFFSET_T_STACK_PTR] = offsetof(struct k_thread,
+						callee_saved.o6),
+#else
+	/* Use a special value so that OpenOCD knows that obtaining the stack
+	 * pointer is not possible on this particular architecture.
+	 */
+#warning Please define ZTA_OFFSET_T_STACK_PTR for this architecture
+	[ZTA_OFFSET_T_STACK_PTR] = ZTA_UNIMPLEMENTED,
+#endif
+	/* Version 0 ends */
+
+	[ZTA_OFFSET_T_NAME] = offsetof(struct k_thread, name),
+	[ZTA_OFFSET_T_ARCH] = offsetof(struct k_thread, arch),
+#if defined(CONFIG_FPU) && defined(CONFIG_FPU_SHARING) && defined(CONFIG_ARM)
+	[ZTA_OFFSET_T_PREEMPT_FLOAT] = offsetof(struct _thread_arch,
+						    preempt_float),
+	[ZTA_OFFSET_T_COOP_FLOAT] = ZTA_UNIMPLEMENTED,
+#elif defined(CONFIG_FPU) && defined(CONFIG_X86)
+#if defined(CONFIG_X86_64)
+	[ZTA_OFFSET_T_PREEMPT_FLOAT] = offsetof(struct _thread_arch, sse),
+#else
+	[ZTA_OFFSET_T_PREEMPT_FLOAT] = offsetof(struct _thread_arch,
+						    preempFloatReg),
+#endif
+	[ZTA_OFFSET_T_COOP_FLOAT] = ZTA_UNIMPLEMENTED,
+#else
+	[ZTA_OFFSET_T_PREEMPT_FLOAT] = ZTA_UNIMPLEMENTED,
+	[ZTA_OFFSET_T_COOP_FLOAT] = ZTA_UNIMPLEMENTED,
+#endif
+	/* Version is still 1, but existence of following elements must be
+	 * checked with _kernel_zta_num_offsets.
+	 */
+};
+extern size_t __attribute__((alias("_kernel_zta_offsets"))) _kernel_openocd_offsets;
+
+__attribute__((used, section(".zta")))
+size_t _kernel_zta_num_offsets = ARRAY_SIZE(_kernel_zta_offsets);
+extern size_t __attribute__((alias("_kernel_zta_num_offsets"))) _kernel_openocd_num_offsets;
+
+__attribute__((used, section(".zta")))
+uint8_t _kernel_zta_size_t_size = (uint8_t)sizeof(size_t);
```

Comment:
this symbol is used outside Zephyr, so require some more work if it is to be renamed.